src/concrete/ml/quantization/quantized_module.py

```diff
@@ -483,12 +487,22 @@ def _clear_forward(
         # The output of a graph must be a QuantizedArray
         assert all(isinstance(elt, QuantizedArray) for elt in output_quantized_arrays)
 
-        results = tuple(
+        q_results = tuple(
             elt.qvalues for elt in output_quantized_arrays if isinstance(elt, QuantizedArray)
         )
-        if len(results) == 1:
-            return results[0]
-        return results
+
+        if self._composition_mapping is not None:
+            q_results = tuple(
+                self.input_quantizers[input_i].quant(
+                    self.output_quantizers[output_i].dequant(q_results[output_i])
+                )
+                for output_i, input_i in self._composition_mapping.items()
+            )
+
+            if len(q_results) == 1:
```

Comment:
can you assert on the shape here ? the input/output shapes should match

Reply:
I've added the checks in the new `_add_requant_for_composition` method (name to be confirmed)